app/code/core/Mage/Contacts/controllers/IndexController.php

```diff
@@ -59,25 +59,24 @@ public function postAction()
             /** @var Mage_Core_Model_Translate $translate */
             $translate->setTranslateInline(false);
             try {
+                if (!$this->_validateFormKey()) {
+                    Mage::throwException($this->__('Invalid Form Key. Please submit your request again.'));
+                }
+
                 $postObject = new Varien_Object();
                 $postObject->setData($post);
 
                 $error = false;
-
                 if (!Zend_Validate::is(trim($post['name']), 'NotEmpty')) {
                     $error = true;
-                }
-
-                if (!Zend_Validate::is(trim($post['comment']), 'NotEmpty')) {
+                } elseif (!Zend_Validate::is(trim($post['comment']), 'NotEmpty')) {
                     $error = true;
-                }
-
-                if (!Zend_Validate::is(trim($post['email']), 'EmailAddress')) {
+                } elseif (!Zend_Validate::is(trim($post['email']), 'EmailAddress')) {
                     $error = true;
                 }
 
                 if ($error) {
-                    throw new Exception();
+                    Mage::throwException($this->__('Unable to submit your request. Please, try again later'));
```

Comment:
I would not use a comma after the word Please, because it doesn't look good. It would mean adding a comma in the phrase above "Please submit your request".